File: java/client/src/main/java/com/github/fernthedev/lightchat/client/ClientCore.java
package com.github.fernthedev.lightchat.client;

import com.github.fernthedev.lightchat.core.Core;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.slf4j.Logger;


@RequiredArgsConstructor
public class ClientCore implements Core {
    @NonNull
    protected Client client;

    @Override
    public boolean isRunning() {
        return client.isRunning();
    }

    @Override
    public Logger getLogger() {
        return client.getLogger();
    }

    @Override
    public String getName() {
        return "Client";
    }



    @Override
    public void shutdown() {
        client.disconnect();
        System.exit(0);
    }
}

File: java/client/src/main/java/com/github/fernthedev/lightchat/client/ClientCore.kt
package com.github.fernthedev.lightchat.client

import com.github.fernthedev.lightchat.core.Core
import org.slf4j.Logger
import kotlin.system.exitProcess

open class ClientCore(@JvmField protected var client: Client) : Core {
    override val isRunning: Boolean
        get() = client.isRunning
    override val logger: Logger
        get() = client.logger
    override val name: String
        get() = "Client"

    override fun shutdown() {
        client.disconnect()
        exitProcess(0)
    }
}